portfolio/src/main/webapp/task.js

```diff
@@ -6,20 +6,123 @@
-   * @param duration: duration of task as Duration class
-   * @param priority: priority level of task, range 1-5 inclusive. 
+   * @param duration: duration of task as String class
+   * @param priority: priority level of task, range 1-5 inclusive.
    */
-  constructor(name, description, duration, priority) {
+  constructor(name, description, duration, taskPriority) {
     this.name = name;
     this.description = description;
     this.duration = duration;
-    this.priority = priority;
+    this.taskPriority = taskPriority;
   }
 }
+
+function createNewTask() {
+  const name = document.getElementById('new-task-name').value;
+  const description = document.getElementById('new-task-description').value;
+  const length = document.getElementById('new-task-estimated-length').value;
+  const lengthUnit =
+      document.getElementById('new-task-estimated-length-unit').value;
+  const priority = document.getElementById('new-task-priority').value;
+
+  const dataArray =
+      [name, description, getDuration(length, lengthUnit), priority];
+
+  if (name == '') {
+    return false;
+  }
+
+  if (parseInt(length) <= 0 || length == '') {
+    return false;
+  }
+
+  const newTask = new Task(
+      name, description, getDuration(length, lengthUnit), parseInt(priority));
+  updateTaskList(newTask);
+}
+
+/** Display Task information from user input. */
+function updateTaskList(newTask, lengthUnit) {
+  const newEventCard = document.createElement('div');
+  newEventCard.classList.add('card');
+
+  const cardBody = document.createElement('div');
+  cardBody.classList.add('card-body');
+  newEventCard.appendChild(cardBody);
+
+  const cardTitle = document.createElement('h4');
+  cardTitle.classList.add('card-title');
+  cardTitle.innerText = newTask.name;
+  cardBody.appendChild(cardTitle);
+
+  const descriptionText = document.createElement('p');
+  descriptionText.classList.add('card-text');
+  descriptionText.innerText = newTask.description;
+  cardBody.appendChild(descriptionText);
+
+  const duration = document.createElement('p');
+  duration.classList.add('card-text');
+  duration.innerText = newTask.duration;
+  cardBody.appendChild(duration);
+
+  const priority = document.createElement('p');
+  priority.classList.add('card-text');
+  priority.innerText = newTask.taskPriority;
+  cardBody.appendChild(priority);
+
+  const deleteButton = document.createElement('button');
+  deleteButton.classList.add('btn');
+  deleteButton.classList.add('btn-danger');
+  deleteButton.innerText = 'Delete this event';
+  cardBody.appendChild(deleteButton);
+
+  const eventList = document.getElementById('new-task-list');
+  eventList.innterHTML = '';
+  eventList.appendChild(newEventCard);
+
+  // The delete button removes the event's card from the UI.
+  deleteButton.onclick = function(newEventCard) {
+    newEventCard.target.closest('div.card').remove();
+  }
+}
+
+/** Return number of minutes from the user's input and unit selection. */
```

Comment:
1. Return => Returns the
2. I think the method name should indicate the time unit.
3. I think the method should return the number of minutes, rather than string.

Reply:
Done.